planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp

```diff
@@ -1039,49 +1036,65 @@ LaneChangeLanesFilteredObjects NormalLaneChange::filterObjects() const
   };
 
   utils::path_safety_checker::filterObjects(
-    target_lane_objects, [&](const PredictedObject & object) {
-      const auto ahead_of_ego = utils::lane_change::is_ahead_of_ego(common_data_ptr_, path, object);
-      return is_within_vel_th(object) || ahead_of_ego;
-    });
+    filtered_by_lanes_objects.target_lane_trailing,
+    [&](const PredictedObject & object) { return is_within_vel_th(object); });
 
   if (lane_change_parameters_->check_objects_on_other_lanes) {
     utils::path_safety_checker::filterObjects(
-      other_lane_objects, [&](const PredictedObject & object) {
+      filtered_by_lanes_objects.other_lane, [&](const PredictedObject & object) {
         const auto ahead_of_ego =
           utils::lane_change::is_ahead_of_ego(common_data_ptr_, path, object);
         return is_within_vel_th(object) && ahead_of_ego;
       });
   }
 
   utils::path_safety_checker::filterObjects(
-    current_lane_objects, [&](const PredictedObject & object) {
+    filtered_by_lanes_objects.current_lane, [&](const PredictedObject & object) {
       const auto ahead_of_ego = utils::lane_change::is_ahead_of_ego(common_data_ptr_, path, object);
       return is_within_vel_th(object) && ahead_of_ego;
     });
 
-  LaneChangeLanesFilteredObjects lane_change_target_objects;
-
+  ExtendedPredictedObjects extended_target_lane_leading_objects;
   const auto is_check_prepare_phase = check_prepare_phase();
-  std::for_each(target_lane_objects.begin(), target_lane_objects.end(), [&](const auto & object) {
-    auto extended_predicted_object = utils::lane_change::transform(
-      object, common_parameters, *lane_change_parameters_, is_check_prepare_phase);
-    lane_change_target_objects.target_lane.push_back(extended_predicted_object);
-  });
+  std::for_each(
```

Comment:
maybe can use a lamda which takes source and target as inputs, instead of repeating the same code 4 times

```
  auto extend_objects = [&](const auto & source, const auto & target) {
    std::for_each(
    source.begin(), source.end()
    [&](const auto & object) {
      auto extended_predicted_object = utils::lane_change::transform(
        object, common_parameters, *lane_change_parameters_, is_check_prepare_phase);
      target.push_back(extended_predicted_object);
    });
  };

  ExtendedPredictedObjects extended_target_lane_leading_objects, extended_target_lane_trailing_objects,
    extended_current_lane_objects, extended_other_lane_objects;
  extend_objects(filtered_by_lanes_objects.target_lane_leading, extended_target_lane_leading_objects);
  extend_objects(filtered_by_lanes_objects.target_lane_trailing, extended_target_lane_trailing_objects);
  extend_objects(filtered_by_lanes_objects.current_lane, extended_current_lane_objects);
  extend_objects(filtered_by_lanes_objects.extended_other_lane_objects, extended_other_lane_objects);
```